plasoscaffolder/frontend/frontend.py

```diff
@@ -0,0 +1,373 @@
+    return_dict = {}
+
+    more_entries = True
+    entry_index = 1
+    while more_entries:
+      # pylint: disable=assignment-from-no-return
+      key = cls.OUTPUT_HANDLER.PromptInfo(
+          '{0:s} [#{1:d}]'.format(question.key_prompt, entry_index))
+      if key in return_dict:
+        cls.OUTPUT_HANDLER.PrintError(
+            'Key value ({0:s}) already entered.'.format(key))
+        continue
+
+      value = cls.OUTPUT_HANDLER.PromptInfo(
+          '{0:s} [#{1:d}]'.format(question.value_prompt, entry_index))
+
+      return_dict[key] = value
+      more_entries = cls.OUTPUT_HANDLER.Confirm('More entries?', abort=False)
+      entry_index += 1
+
+    question.ValidateAnswer(return_dict)
+    return return_dict
+
+  @classmethod
+  def _AskListQuestion(
+      cls, question: scaffolder_interface.BaseQuestion) -> List[object]:
+    """Ask the user a question and return a list back.
+
+    Args:
+      question (scaffolder_interface.BaseQuestion): the question to ask.
+
+    Returns:
+      list: the answer as supplied by the user.
+    """
+    return_list = []
+
+    more_entries = True
+    entry_index = 1
+    while more_entries:
+      # pylint: disable=assignment-from-no-return
+      value = cls.OUTPUT_HANDLER.PromptInfo(
+          'Value to add [#{0:d}]'.format(entry_index))
+      if not value:
+        cls.OUTPUT_HANDLER.PrintError('Empty value, not adding.')
+      else:
+        return_list.append(value)
+
+      more_entries = cls.OUTPUT_HANDLER.Confirm('Add more values?', abort=False)
+      entry_index += 1
+
+    question.ValidateAnswer(return_list)
+    return return_list
+
+  @classmethod
+  def _AskQuestion(cls, question: scaffolder_interface.BaseQuestion) -> Type:
+    """Ask the user a question and return an answer back.
+
+    Args:
+      question (scaffolder_interface.BaseQuestion): the question to ask.
+
+    Returns:
+      object: an object, whos type is defined in question.type.
+
+    Raises:
+      errors.WrongCliInput: when an unsupported question type is
+          encountered.
+    """
+    cls.OUTPUT_HANDLER.PrintNewLine()
+    cls.OUTPUT_HANDLER.PrintInfo(question.prompt)
+
+    if question.TYPE == str:
+      return cls._AskStringQuestion(question)
+
+    if question.TYPE == dict:
+      return cls._AskDictQuestion(question)
+
+    if question.TYPE == list:
+      return cls._AskListQuestion(question)
+
+    # TODO: Add support for other types of questions.
+    raise errors.WrongCliInput(
+        'Question type {0:s} not supported.'.format(str(question.TYPE)))
+
+  @classmethod
+  def _AskStringQuestion(
+      cls, question: scaffolder_interface.BaseQuestion) -> str:
+    """Ask the user a question and return a string answer back.
+
+    Args:
+      question (scaffolder_interface.BaseQuestion): the question to ask.
+
+    Returns:
+      str: the answer as supplied by the user.
+    """
+    # pylint: disable=assignment-from-no-return
+    value = cls.OUTPUT_HANDLER.PromptInfo('Value')
```

Comment:
Unless I'm misunderstanding, this should be question.prompt, not 'Value'

Reply:
OK, I get it now, the prompt is displayed, then we prompt again with the string 'Value'